controllers/front/expresscheckout.php

```diff
@@ -94,30 +94,30 @@ private function _expressCheckoutInitial()
 			$nvp_request .= '&L_PAYMENTREQUEST_0_NAME'.$i.'='.urlencode($product['name']).
 					'&L_PAYMENTREQUEST_0_NUMBER'.$i.'='.urlencode((int)$product['id_product']).
 					'&L_PAYMENTREQUEST_0_DESC'.$i.'='.urlencode(strip_tags(Tools::truncate($product['description_short'], 80))).
-					'&L_PAYMENTREQUEST_0_AMT'.$i.'='.urlencode((float)$product['price_wt']).
+					'&L_PAYMENTREQUEST_0_AMT'.$i.'='.urlencode(number_format($product['price_wt'],2)).
 					'&L_PAYMENTREQUEST_0_QTY'.$i.'='.urlencode((int)$product['cart_quantity']);
 					
 			$total += ((float)$product['price_wt'] * (int)$product['cart_quantity']);
 			
 			$i++;
 		}
+		$shipping = (float)$this->context->cart->getTotalShippingCost();
+		$cupon = - $total + $totalToPay - $shipping;
 
 		if ($cart_discount = $this->context->cart->getOrderTotal(false, Cart::ONLY_DISCOUNTS))
 		{
-			$nvp_request .= '&L_PAYMENTREQUEST_0_NAME'.$i.'='.urlencode($this->paypal_mx->l('Coupon')).
-					'&L_PAYMENTREQUEST_0_AMT'.$i.'='.urlencode(number_format($cart_discount,2)).
+			$nvp_request .= '&L_PAYMENTREQUEST_0_NAME'.$i.'='.urlencode($this->paypal_mx->l('Cupón')).

```

Comment:
Hi Thomas,
No, this value wil never change, this module is designed for Mexican
Merchants,
Regards,


*Ing. Miguel Barrientos*
Project Manager

Pengo Stores
Apasionados del e-commerce
www.pengostores.com
Oficina: +52 55 9628-2118
Móvil: +52 1 55 4678-6845

2015-05-22 7:41 GMT-05:00 Thomas N <notifications@github.com>:

> In controllers/front/expresscheckout.php
> <https://github.com/PrestaShop/paypalmx/pull/10#discussion_r30891118>:
>
> >
> >  		if ($cart_discount = $this->context->cart->getOrderTotal(false, Cart::ONLY_DISCOUNTS))
> >  		{
> > -			$nvp_request .= '&L_PAYMENTREQUEST_0_NAME'.$i.'='.urlencode($this->paypal_mx->l('Coupon')).
> > -					'&L_PAYMENTREQUEST_0_AMT'.$i.'='.urlencode(number_format($cart_discount,2)).
> > +			$nvp_request .= '&L_PAYMENTREQUEST_0_NAME'.$i.'='.urlencode($this->paypal_mx->l('Cupón')).
>
> Does this value change depending of the current language ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/PrestaShop/paypalmx/pull/10/files#r30891118>.
>